### DISCUSSION TITLE
[General] Roadmap for GLPI Agent 1.5

### DISCUSSION THREAD

**Discussion by AndiMb** [glpi-project/glpi-agent#373] [General]
### Your question

Hi,

I just want to ask, if there is any date for publishing Version 1.5 of the GLPI Agent. I know, there are nightly builds, but it would be nice to have a new stable version without the bugs of 1.4.

Regards,
Andreas

**Comment by g-bougard** [comment]
Hi @AndiMb 
no date for now, we are in the process of renewing our EV code signing certificate for MSI packages but our provider was failing to process our request. We decided to switch to another provider and so this may again take some time before we are able to produce an official release with correctly signed packages.

**Reply by tiangao88** [reply]
Hello @g-bougard, what are the news on the code signing certificate renewal?
Can we expect a 1.5 release any time soon?

**Reply by g-bougard** [reply]
Hi @tiangao88

no really good news from our provider. EV SSL code-signing certificates providers seems to not really care about their customers. This really sounds like a bad joke for us...

We may fallback on a non EV certificate for the 1.5 version release which seems easier to obtain. And we would then come back to an EV one for, let's day, 1.6 release. This would really be better than worse... I'll check that alternative next week with my boss.

**Comment by kintaro1981** [comment]
@AndiMb actually we are with fusion agent 2.5.2. Are there any bugs in glpi agent 1.4 that are worth waiting for 1.5?

**Reply by Dan-Bradd** [reply]
Yes there is some bugs in the 1.4 i've noticed.
The last one who work correctly and who is used in the compagny i work for is : "fusioninventory-agent_windows-x64_2.6.exe"
but it is not secure like the GLPI agent 1.4.
In GLPI agent 1.4, with the msi installation file or the vbs file it is the same BUG : it works once, the inventory is done at the end of the installation and after agent 1.4 don't contact the server.
Why ? because if you use GLPI with a windows task and not a service task who is set in the GLPI server don't use it !

i've tried to put value regkey to 1 for "listen" but it is the same.
I've checked many times but it is the same issue according to the fusioninventory-agent_windows-x64_2.6.exe who work perfectly.

it's like the GLPI AGENT 1.4 is not a completely finished product who has not been completely tested.

My apologizes for developpers who work very well but unfortunately it doesntt work with a windows tasks.

So we need to create in fact the windows task and it doesn't work anyway because maybe there is no link in the registry between :

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Schedule\TaskCache\Tasks\ID value

& 

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Schedule\TaskCache\Tree\FusionInventory-Agent\Id\ID value


please confirm it could work by this way and it would be enough secure.

THANKS to all

**Reply by Dan-Bradd** [reply]
if you want to use the FusionInventory-Agent 2.6, you have to know there is no msi file for the moment but it is the latest stable version.

https://github.com/fusioninventory/fusioninventory-agent/releases

**Reply by g-bougard** [reply]
Hi @Dan-Bradd 

> it's like the GLPI AGENT 1.4 is not a completely finished product who has not been completely tested.

Are you kidding ? I worked on GLPI-Agent in the continuity of FI Agent for which I was an active maintainer since FI 2.3.18. They are both tested at the same level.
You should better understand such kind of software won't never be completed as systems evolves and inventory agents need to evolves at the same time.

> In GLPI agent 1.4, with the msi installation file or the vbs file it is the same BUG : it works once, the inventory is done at the end of the installation and after agent 1.4 don't contact the server.
> Why ? because if you use GLPI with a windows task and not a service task who is set in the GLPI server don't use it !

Here you're talking about a bug but never opened any issue. And to be honest I don't understand your explanation. I can only think you did something wrong during the installation.

> i've tried to put value regkey to 1 for "listen" but it is the same.

Don't expect an agent which is not running as a daemon to listen for http requests. This is normal and this is not a bug.

> My apologizes for developpers who work very well but unfortunately it doesntt work with a windows tasks.

You're wrong. We have a lot of customer who use installation as windows tasks and this works as expected.

I think you simply failed to configure your agent in the correct way. And so if you need/want help, please open a dedicated Q&A discussion.

**Reply by Dan-Bradd** [reply]
Hello g-bougard,
You are right the GLPI agent 1.4 seems finally to work perfectly, but in fact if we choose the msi file to install, we should not to forget to use the usefull command line to create a windows task at the same time...normaly the command line should look like something like this :

msiexec.exe /i "%userprofile%\AppData\Local\Temp\glpi-agent_windows-x64.msi"  \quiet SERVER='http://yourGLPIserveradress/glpi/plugins/fusioninventory/'  EXECMODE=2 LAZY=0 NO_HTTPD=1 TASK_FREQUENCY=daily TASK_DAILY_MODIFIER=1 ADDLOCAL=ALL RUNNOW=1 NO_SSL_CHECK=1 PROXY='http://yourproxyserveradress'
Scheduling: DEL /Q /F "%userprofile%\AppData\Local\Temp\glpi-agent_windows-x64.msi"
schtasks /Change /TN "GLPI Agent" /ST "09:15"

Unfortunately the hour who is set in the windows task 9:15 is random and it is not set to office hours, but the Agent works.

**Reply by g-bougard** [reply]
Please, open another Q&A thread if you really need an answer as this is not the right place to discuss about such problem.
